### PR TITLE
Transpile ES6 code through Babel in Webpack 4

### PR DESCRIPTION
**Pull request by jesusbotella**
This PR enables us to write ES6 code in Carto.js, by later transpiling it when making the production bundle.

And I did the update to Webpack 4 as well because I saw it easy 🤷‍♂️ 

### ⚠️ WARNING  ⚠️ 
This PR has to be merged in conjunction with https://github.com/CartoDB/cartodb/pull/14117

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,7 +25,8 @@
     "Ruben Moya <ruben@carto.com>",
     "Jesus Arroyo Torrens <jarroyo@carto.com>",
     "Iago Lastra <iago@carto.com>",
-    "Elena Torró <elena@carto.com>"
+    "Elena Torró <elena@carto.com>",
+    "Jesús Botella <jbotella@carto.com>"
```
[the remark this answers is not in the document]
Comment (rjimenezda) [reply package.json]: 😂

File: package.json [hunk 2]
```diff
@@ -50,8 +51,12 @@
     "whatwg-fetch": "^2.0.3"
   },
   "devDependencies": {
+    "babel-core": "^6.26.3",
+    "babel-loader": "^7.1.4",
+    "babel-plugin-transform-object-rest-spread": "^6.26.0",
+    "babel-preset-env": "^1.7.0",
     "babel-preset-es2015": "~6.24.1",
-    "babelify": "~7.3.0",
+    "babelify": "^7.3.0",
```
Comment (rjimenezda) [inline package.json]: 👍

**Comment by jesusbotella** [comment]
I think I did a bundle size comparison and there wasn't a big difference in terms of size :(

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (elenatorro) [reply package.json]: 🕵️

[the remark this answers is not in the document]
Comment (jesusbotella) [reply package.json]: Hey! What's going on here? 😂

[the remark this answers is not in the document]
Comment (rjimenezda) [reply package.json]: but what happened 🤷‍♂️

**Comment by rjimenezda** [comment]
That's fine, we're adding support for ES6 so I was worried we might have some extra boilerplate.

300kb is huge though 🤔

**Comment by jesusbotella** [comment]
Yes, I passed the Webpack Bundle Analyzer, and it was mostly jQuery, Backbone, and Underscore.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (jesusbotella) [reply package.json]: tenor

**Pull request merged**